File: Python/Week_2/day1_functions.py
# Week 2 - Day 1 - Functions in Python

# Basic function - no parameters, no return
def greet():
    print("Hello, welcome to Python functions!")

greet()   # calling the function

# Function with parameters
def greet_user(name):
    print("Hello,", name)

greet_user("Ali")

# Function with return value
def add(a, b):
    return a + b

result = add(5, 3)
print("Sum:", result)

# Function with default parameter value
def greet_with_default(name="Guest"):
    print("Hello,", name)

greet_with_default()          # uses default "Guest"
greet_with_default("Sara")    # overrides default

# Function with multiple return values
def calculate(a, b):
    return a + b, a - b, a * b

sum_val, diff_val, mult_val = calculate(10, 4)
print("Sum:", sum_val, "Diff:", diff_val, "Mult:", mult_val)

# Function calling another function
def square(n):
    return n * n

def sum_of_squares(a, b):
    return square(a) + square(b)

print("Sum of squares:", sum_of_squares(2, 3))

File: Java/Week_2/Day1Functions.java
package Week_2;

public class Day1Functions {
    
    // Basic method - no parameters, no return (void)
    static void greet() {
        System.out.println("Hello, welcome to Java methods!");
    }
    
    // Method with parameters
    static void greetUser(String name) {
        System.out.println("Hello, " + name);
    }
    
    // Method with return value
    static int add(int a, int b) {
        return a + b;
    }
    
    // Method calling another method
    static int square(int n) {
        return n * n;
    }
    
    static int sumOfSquares(int a, int b) {
        return square(a) + square(b);
    }
    
    public static void main(String[] args) {
        
        greet();
        greetUser("Ali");
        
        int result = add(5, 3);
        System.out.println("Sum: " + result);
        
        System.out.println("Sum of squares: " + sumOfSquares(2, 3));
    }
}